src/js/components/search/filters/timePeriod/TimePeriod.jsx

```diff
@@ -57,10 +56,10 @@ export default class TimePeriod extends React.Component {
         // validate that dates are provided for both fields and the end dates
         // don't come before the start dates
 
-        const rng = Object.assign({}, this.state);
+        const err = Object.assign({}, this.state);
 
         const output = {
-            range: rng
+            error: err
```

Comment:
I still don't understand what this is happening here. In the `constructor`, the state is defined as having a child object named `error`. But here, you're cloning a new instance of the state, and then setting the value of `output`'s `error` key to the _entire_ cloned state...which incidentally, gets overwritten anyway in lines 68-95.